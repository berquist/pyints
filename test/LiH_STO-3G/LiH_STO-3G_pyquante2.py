#!/usr/bin/env python

from __future__ import print_function

import numpy as np

import pyquante2

with open('LiH.xyz') as molfile:
    mollines = molfile.readlines()[2:]

mol = pyquante2.geo.molecule.read_xyz_lines(mollines,
                                            units='Angstrom',
                                            charge=0,
                                            multiplicity=1,
                                            name='LiH')

del mollines

print(mol)

mol_basis = pyquante2.basisset(mol, 'STO-3G')

print(mol_basis)
# print('There are {} basis functions.'.format(len(mol_basis)))
print('There are {} shells.'.format(len(mol_basis.shells)))
print(mol_basis.shells)

solver = pyquante2.uhf(mol, mol_basis)
solver.converge(tol=1e-10, maxiters=1000)

print(solver)

i1 = solver.i1
# print('Converged overlap matrix:')
# print(i1.S)
# print('Converged kinetic energy matrix:')
# print(i1.T)
# print('Converged potential energy matrix:')
# print(i1.V)

C = solver.orbsa
D = np.dot(C[:, :mol.nocc()], C[:, :mol.nocc()].transpose())

# Save the one-electron integral matrices to disk.
np.savetxt('pyquante2.S.txt', i1.S)
np.savetxt('pyquante2.T.txt', i1.T)
np.savetxt('pyquante2.V.txt', i1.V)
np.savetxt('pyquante2.D.txt', D)

# Save the energies to disk.
np.savetxt('pyquante2.total_energy.txt', np.array([solver.energy]))
np.savetxt('pyquante2.nuclear_repulsion_energy.txt', np.array([mol.nuclear_repulsion()]))
