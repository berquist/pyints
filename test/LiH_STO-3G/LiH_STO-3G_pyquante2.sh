#!/usr/bin/env bash

python LiH_STO-3G_pyquante2.py | tee LiH_STO-3G_pyquante2.out
