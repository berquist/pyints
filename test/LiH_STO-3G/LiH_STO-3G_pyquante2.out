Stoichiometry = HLi,  Charge = 0,  Multiplicity = 1
3 Li     0.000000     0.000000     0.000000
1 H     0.000000     0.000000     3.013924
cgbf((0.0, 0.0, 0.0), (0, 0, 0), [16.119575, 2.9362007, 0.7946505], [0.15432897, 0.53532814, 0.44463454])
cgbf((0.0, 0.0, 0.0), (0, 0, 0), [0.6362897, 0.1478601, 0.0480887], [-0.09996723, 0.39951283, 0.70011547])
cgbf((0.0, 0.0, 0.0), (1, 0, 0), [0.6362897, 0.1478601, 0.0480887], [0.15591627, 0.60768372, 0.39195739])
cgbf((0.0, 0.0, 0.0), (0, 1, 0), [0.6362897, 0.1478601, 0.0480887], [0.15591627, 0.60768372, 0.39195739])
cgbf((0.0, 0.0, 0.0), (0, 0, 1), [0.6362897, 0.1478601, 0.0480887], [0.15591627, 0.60768372, 0.39195739])
cgbf((0.0, 0.0, 3.0139239791845247), (0, 0, 0), [3.42525091, 0.62391373, 0.1688554], [0.15432897, 0.53532814, 0.44463454])
There are 4 shells.
[S shell: [16.119575, 2.9362007, 0.7946505], [0.15432897, 0.53532814, 0.44463454], S shell: [0.6362897, 0.1478601, 0.0480887], [-0.09996723, 0.39951283, 0.70011547], P shell: [0.6362897, 0.1478601, 0.0480887], [0.15591627, 0.60768372, 0.39195739], S shell: [3.42525091, 0.62391373, 0.1688554], [0.15432897, 0.53532814, 0.44463454]]
UHF Hamiltonian
Stoichiometry = HLi,  Charge = 0,  Multiplicity = 1
3 Li     0.000000     0.000000     0.000000
1 H     0.000000     0.000000     3.013924
Basis set: STO-3G, Nbf: 6
Status: Converged = True
  0   -3.1425455
  1   -7.8185992
  2   -7.8610138
  3   -7.8618972
  4   -7.8619938
  5   -7.8620178
  6   -7.8620244
  7   -7.8620262
  8   -7.8620267
  9   -7.8620269
 10   -7.8620269
 11   -7.8620269
 12   -7.8620269
 13   -7.8620269
 14   -7.8620269
